modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -182,7 +182,7 @@ object CurrencySnapshotCreator {
             stateProof,
             currentEpochProgress,
             dataApplicationAcceptanceResult.map(_.dataApplicationPart),
-            messagesAcceptanceResult.accepted.toSortedSet.some,
+            Option.when(snapshotInfo.lastMessages.nonEmpty)(messagesAcceptanceResult.accepted.toSortedSet),
```

Comment:
Start adding messages to the CurrencyIncrementalSnapshot the moment currencyMessages state is non empty. It will also set the list as empty after first message appears.